### PR TITLE
Using user secrets for ConnectionString

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

**Comment by hishamco** [comment]
@thiennn can you merge this .. and look to my comments in all the issues, because I passed through almost the issues in the repo, some of them need to be closed to duplicate or needs your review

I knew it needs a time to close all of them, but let us struggle to finalize & schedule them

File: README.md [hunk 1]
```diff
@@ -27,7 +27,21 @@
...
+    ```
+    - User Secrets Tool for Visual Studio
+        1. Right click the SimplCommerce.WebHost 
+        2. Choose "Manage User Secrets"
+        3. Write Your secrets in `secrets.json` as the following:
+        ```
+        {
+            "ConnectionStrings:DefaultConnection": "Your ConnectionString"
+        }
+        ```
```
Comment (thiennn) [inline README.md]: I think we shouldn't make it more complicated to user. I prefer to make thing as simple as possible. Advanced user will know how to use Secrets if they want to

File: src/SimplCommerce.WebHost/Program.cs [hunk 2]
```diff
@@ -29,6 +29,7 @@ private static void SetupConfiguration(WebHostBuilderContext hostingContext, ICo
                 .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
 
             configBuilder.AddEnvironmentVariables();
+            configBuilder.AddUserSecrets<Program>();
```
Comment (thiennn) [inline src/SimplCommerce.WebHost/Program.cs]: This fine

File: src/SimplCommerce.WebHost/appsettings.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿{
   "ConnectionStrings": {
-    "DefaultConnection": "Server=.\\SQLEXPRESS;Initial Catalog=SimplCommerce;Integrated Security=True;MultipleActiveResultSets=True"
+    "DefaultConnection": "Your ConnectionString"
```
Comment (thiennn) [inline src/SimplCommerce.WebHost/appsettings.json]: I don't think it is a good idea to change this value. It eventually will be overrided.

File: README.md (hunk 1 above)
Comment (hishamco) [reply README.md]: This not complicated while I update the `README.md`, furthermore this is good for security, the user shouldn't expose any kind of secrets

File: src/SimplCommerce.WebHost/appsettings.json (hunk 3 above)
Comment (hishamco) [reply src/SimplCommerce.WebHost/appsettings.json]: We can make it as its for the novice users, again the `README.md` will describes everything to make the platform up & running

Comment (tdavisjr) [reply src/SimplCommerce.WebHost/appsettings.json]: SQL Server Express requires a separate install as it doesn't come with the VS installation. For ease if use so new comers can get up and running quickly, please consider MS SQL LocalDB as it comes bundled with VS and it works just as good as SQL Server Express.

**Comment by thiennn** [comment]
How to use Secret Manager is added here https://github.com/simplcommerce/docs/blob/master/docs/development-tips.md

File: src/SimplCommerce.WebHost/appsettings.json (hunk 3 above)
Comment (hishamco) [reply src/SimplCommerce.WebHost/appsettings.json]: I agree with what @tdavisjr is suggested, can we change the default connectionString to LocalDB?

**Comment by hishamco** [comment]
@thiennn it would be nice if we mention the User Secrets in `README.md` as I did in the PR

**Comment by thiennn** [comment]
I think too much information in the readme is not good

**Comment by hishamco** [comment]
But that is essential to make the app up & running

**Comment by thiennn** [comment]
It's easier for users to make the app running without concerning about Secret Manager and how does it work. 

Using secret manager is not required to make the app up and running. Of course they can use it if they prefer to.

**Pull request closed without merging**